src/plugins/ros2/irscan_thread.h

```diff
@@ -0,0 +1,76 @@
+
+/***************************************************************************
+ *  laserscan_thread.cpp - Thread to exchange IR Sensor data
+ *
+ *  Created: Mon Jul 03 13:41:18 2012
+ *  Copyright  2023 Saurabh Borse, Tim Wendt
+ ****************************************************************************/
+
+/*  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Library General Public License for more details.
+ *
+ *  Read the full text in the LICENSE.GPL file in the doc directory.
+ */
+
+#ifndef _PLUGINS_ROS_IRSCANNER_THREAD_H_
+#define _PLUGINS_ROS_IRSCANNER_THREAD_H_
+
+#include <aspect/blackboard.h>
+#include <aspect/blocked_timing.h>
+#include <aspect/configurable.h>
+#include <aspect/logging.h>
+#include <blackboard/interface_listener.h>
+#include <blackboard/interface_observer.h>
```

Comment:
the interface_observer, mutex and logging includes are unnecessary (usage of the logging aspect in general, as you do not log anything).